# Issue 32251: Update numpy to 1.21.2

archive/issues_032251.json:
```json
{
    "body": "CC:  dimpase fbissey arojas slelievre\n\n(split out from #32105)\n\n\u200bhttps://github.com/numpy/numpy/releases\n\n\"The Python versions supported for this release are 3.7-3.9. The 1.21.x series is compatible with Python 3.10.0rc1 and Python 3.10 will be officially supported after it is released.\"\n\n\n\nIssue created by migration from https://trac.sagemath.org/ticket/32488\n\n",
    "created_at": "2021-09-07T22:19:17Z",
    "labels": [
        "packages: standard",
        "major",
        "enhancement"
    ],
    "title": "Update numpy to 1.21.2",
    "type": "issue",
    "url": "https://github.com/sagemath/sagetest/issues/32251",
    "user": "mkoeppe"
}
```
CC:  dimpase fbissey arojas slelievre

(split out from #32105)

â€‹https://github.com/numpy/numpy/releases

"The Python versions supported for this release are 3.7-3.9. The 1.21.x series is compatible with Python 3.10.0rc1 and Python 3.10 will be officially supported after it is released."



Issue created by migration from https://trac.sagemath.org/ticket/32488





---

archive/issue_comments_460522.json:
```json
{
    "body": "New commits:",
    "created_at": "2021-09-07T22:28:19Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460522",
    "user": "mkoeppe"
}
```

New commits:



---

archive/issue_comments_460523.json:
```json
{
    "body": "Changing status from new to needs_review.",
    "created_at": "2021-09-07T22:28:19Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460523",
    "user": "mkoeppe"
}
```

Changing status from new to needs_review.



---

archive/issue_comments_460524.json:
```json
{
    "body": "Tests look good; and even the usual cygwin segfaults on plotting are gone. (But we had sporadic success of the cygwin CI before, so we cannot be sure yet that this ticket fixed it.)\n\nReady for review.",
    "created_at": "2021-09-09T17:40:40Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460524",
    "user": "mkoeppe"
}
```

Tests look good; and even the usual cygwin segfaults on plotting are gone. (But we had sporadic success of the cygwin CI before, so we cannot be sure yet that this ticket fixed it.)

Ready for review.



---

archive/issue_comments_460525.json:
```json
{
    "body": "Changing priority from major to critical.",
    "created_at": "2021-09-09T17:40:40Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460525",
    "user": "mkoeppe"
}
```

Changing priority from major to critical.



---

archive/issue_comments_460526.json:
```json
{
    "body": "I wonder whether this plays along with SciPy we ship well, or we should bump SciPy too.",
    "created_at": "2021-09-09T18:15:59Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460526",
    "user": "dimpase"
}
```

I wonder whether this plays along with SciPy we ship well, or we should bump SciPy too.



---

archive/issue_comments_460527.json:
```json
{
    "body": "OK, SciPy will get updated on #32105",
    "created_at": "2021-09-09T18:47:27Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460527",
    "user": "dimpase"
}
```

OK, SciPy will get updated on #32105



---

archive/issue_comments_460528.json:
```json
{
    "body": "Our scipy 1.6.3 declares that it is compatible with:\n\n```\n    numpy>=1.16.5,<1.23.0\n```\n\n... so yes, our current version is compatible with this update.",
    "created_at": "2021-09-09T20:35:16Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460528",
    "user": "mkoeppe"
}
```

Our scipy 1.6.3 declares that it is compatible with:

```
    numpy>=1.16.5,<1.23.0
```

... so yes, our current version is compatible with this update.



---

archive/issue_comments_460529.json:
```json
{
    "body": "lgtm",
    "created_at": "2021-09-09T21:10:11Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460529",
    "user": "dimpase"
}
```

lgtm



---

archive/issue_comments_460530.json:
```json
{
    "body": "Changing status from needs_review to positive_review.",
    "created_at": "2021-09-09T21:10:11Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460530",
    "user": "dimpase"
}
```

Changing status from needs_review to positive_review.



---

archive/issue_comments_460531.json:
```json
{
    "body": "Thanks!",
    "created_at": "2021-09-09T22:55:10Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460531",
    "user": "mkoeppe"
}
```

Thanks!



---

archive/issue_comments_460532.json:
```json
{
    "body": "Resolution: fixed",
    "created_at": "2021-09-15T22:06:20Z",
    "issue": "https://github.com/sagemath/sagetest/issues/32251",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/32251#issuecomment-460532",
    "user": "vbraun"
}
```

Resolution: fixed
